internal/cmd/agent/deployer/monitor/updatestatus.go

```diff
@@ -263,7 +285,11 @@ func modified(ctx context.Context, c client.Client, logger logr.Logger, plan des
 			return sortKey(result[i]) < sortKey(result[j])
```

Comment:
We need to keep sorting result, since the keys in `plan`'s `Create`, `Delete`, `Modified` could be different.